src/lib.rs

```diff
@@ -20,18 +20,21 @@ use std::ptr;
 ///
 /// Guarantees almost nothing else, especially that values stored or loaded
 /// are actually valid values for type `T`.
-pub struct RaceBuf<T>(Vec<T>);
+pub struct RaceBuf<T>(std::cell::UnsafeCell<Vec<T>>);
```

Comment:
`UnsafeCell` means that you are allowed to violate the normal rules under certain conditions. Just using `volatile_write` doesn't mean anything. You could also do `Vec<Volatile<T>>` if you use the `Volatile` from the embedded crates. This'll save you a lot of headaches.